modules/elements-color-picker/assets/js/editor/commands/show-swatches.js

```diff
@@ -98,31 +111,57 @@ export class ShowSwatches extends CommandBase {
 
 			// Handle background images.
 			if ( isBgImage ) {
-				this.addTempBackgroundImage( this.container.getSetting( control ) );
+				this.addTempBackgroundImage( container.getSetting( control ) );
 				return;
 			}
 
-			let value = this.container.getSetting( control );
+			let value = container.getSetting( control );
+			const globalValue = container.globals.get( control );
 
-			if ( value && ! Object.values( this.colors ).includes( value ) ) {
-				// If it's a global color, it will return a css variable which needs to be resolved to a HEX value.
-				const pattern = /var\(([^)]+)\)/i;
-				const matches = value.match( pattern );
+			// Extract global value if present.
+			if ( globalValue ) {
+				const matches = globalValue.match( /id=(.+)/i );
 
+				// Build the global color CSS variable & resolve it to a HEX value.
+				// It's used instead of `$e.data.get( globalValue )` in order to avoid async/await hell.
```

Comment:
```suggestion
				// It's used instead of `$e.data.get( globalValue )` in order to avoid async/await.
```

Reply:
That's the term used to describe this:
https://medium.com/@7genblogger/escaping-so-called-async-await-hell-in-node-js-b5f5ba5fa9ca